config/config.go

```diff
@@ -115,3 +127,16 @@ type WebhookProject struct {
 	Secret             string   `json:"-"`
 	SkipSignatureCheck bool     `json:"skipSignatureCheck" default:"false"`
 }
+
+// OAuthClientCredentials are used for issuing access tokens
+type OAuthClientCredentials struct {
+	ID     string `yaml:"id"`
+	Secret string `yaml:"secret"`
+}
+
+// ServiceAuthConfig holds the authentication configuration for a particular service
+type ServiceAuthConfig struct {
+	Clients    []OAuthClientCredentials `yaml:"clients"`
+	HMACSecret string                   `yaml:"hmacSecret"`
```

Comment:
must include ``json:"-"`` so that these are exposed when marshalled to json via the `/config` endpoint